File: src/SpaJS.A.Web/app/domain/Customer.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var app;
(function (app) {
    var domain;
    (function (domain) {
        var Customer = (function (_super) {
            __extends(Customer, _super);
            function Customer(firstName, surname, active, address, id) {
                _super.call(this);
                this.firstName = firstName;
                this.surname = surname;
                this.active = active;
                this.address = address;
                this.id = id;
                this.id = id;
                this.firstName = firstName;
                this.surname = surname;
                this.active = active;
                this.address = address;
            }
            return Customer;
        })(app.domain.EntityBase);
        domain.Customer = Customer;
    })(domain = app.domain || (app.domain = {}));
})(app || (app = {}));

File: src/SpaJumpstart.Domain/Entities/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SpaJumpstart.Domain.Entities
{
    /// <summary>
    /// Customer Details
    /// </summary>
    public class Customer : EntityBase, IValidatableObject
    {
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string Surname { get; set; }
        public string Telephone { get; set; }
        public string MobilePhone { get; set; }
        public bool Active { get; set; }
        public DateTime InceptionDate { get; set; }

        public string ApplicationUserId { get; set; }
        public virtual ApplicationUser ApplicationUser { get; set; }

        [ForeignKey("Id")]
        public virtual Address Address { get; set; }

        //This adds a rowversion id to the table for concurrency
        //[Timestamp]
        //public byte[] RowVersion { get; set; }

        #region Validation
        //Belt and Braces validation

        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            var results = new List<ValidationResult>();
            if (FirstName.Contains("\n"))
            {
                results.Add(
                    new ValidationResult("Newline character is illegal",
                        new[] { "FirstName" }));
            }
            if (Surname.Contains("\n"))
            {
                results.Add(
                    new ValidationResult("Newline character is illegal",
                        new[] { "Surname" }));
            }

            if (string.IsNullOrEmpty(FirstName))
            {
                results.Add(
                    new ValidationResult("FirstName is required",
                        new[] { "FirstName" }));
            }
            if (string.IsNullOrEmpty(Surname))
            {
                results.Add(
                    new ValidationResult("Surname is required",
                        new[] { "Surname" }));
            }
            if (Address == null)
            {
                results.Add(
                    new ValidationResult("Address is required",
                        new[] { "Address" }));
            }
            return results;
        }

        #endregion
    }
}

